osu.Game/Database/BeatmapDatabase.cs

```diff
@@ -186,7 +186,7 @@ private BeatmapSetInfo getBeatmapSet(string path)
                 {
                     hash = input.GetMd5Hash();
                     input.Seek(0, SeekOrigin.Begin);
-                    path = Path.Combine(@"beatmaps", hash.Remove(1), hash.Remove(2), hash);
+                    path = Path.Combine(@"beatmaps", hash.Remove(1), hash.Remove(2), hash+".osz");

```

Comment:
spaces before and after `+`.

Also the `.osz` extension should be a `const` somewhere, as you've used it in three? different places here.